flake.nix

```diff
@@ -0,0 +1,154 @@
+{
+  description = "Nix package for centrifuge-chain";
+
+  inputs = {
+    nixpkgs.url = github:NixOS/nixpkgs/nixos-21.11;
+    flake-utils = {
+        url = github:numtide/flake-utils;
+        inputs.nixpkgs.follows = "nixpkgs";
+    };
+    gitignore = {
+      url = github:hercules-ci/gitignore.nix;
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    fenix = {
+      url = github:nix-community/fenix;
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = inputs :
+    inputs.flake-utils.lib.eachDefaultSystem (system:
+        let
+          pkgs = inputs.nixpkgs.legacyPackages.${system};
+
+          cargoTOML = builtins.fromTOML (builtins.readFile ./Cargo.toml);
```

Comment:
@branan Now I read from file some params, I had to diver deeper on nix-how-to, but we figured it out.